EGE/Gen/EGE/A06.py

```diff
@@ -0,0 +1,435 @@
+        ]
+        code_block_parts = [
+            'for', iteration_variable, 1, array_length, first_loop_body,
+            'for', iteration_variable, 1, array_length, second_loop_body,
+            '#', { 'C': c_language_comment },
+        ]
+        code_block = make_block(code_block_parts)
+        lang_table = table(code_block, [ [ 'Basic', 'Alg' ], [ 'Pascal', 'C' ] ])
+        case = self.rnd.pick([
+            { 'name': 'положительные',   'test': lambda x: x > 0  },
+            { 'name': 'отрицательные',   'test': lambda x: x < 0  },
+            { 'name': 'неотрицательные', 'test': lambda x: x >= 0 },            
+        ])
+        self.text = f'''
+Значения двух массивов A и B с индексами от 1 до {array_length}
+задаются при помощи следующего фрагмента программы: {lang_table}
+Какое количество элементов массива B[1..{array_length}] будет принимать
+{case['name']} значения после выполнения данной программы?'''
+
+        array_B = code_block.run_val('B')
+        correct = len(list(filter(lambda x: case['test'](x), array_B.values())))
+        errors = [
+            correct + 1, correct - 1, array_length - correct, array_length - correct + 1, array_length - correct - 1
+        ]
+        self.set_variants([ correct ] + self.rnd.pick_n(3, errors))
+        return self
+
+class FindMinMax(SingleChoice):
+    def generate(self):
+        array_length = self.rnd.in_range(50, 100)
+        # Нужно гарантировать единственные максимум и минимум.
+        value = self.rnd.in_range(array_length // 2 + 1, array_length - 1)
+        iteration_variable = self.rnd.index_var()[0]
+        factor = [ '-' ] + self.rnd.shuffle([ iteration_variable, value ])
+        indexB, indexA = self.rnd.shuffle([
+            iteration_variable, [ '-', array_length + 1, iteration_variable ]
+        ])
+        first_loop_body = [
+            '=', [ '[]', 'A', iteration_variable ], [ '*', factor, factor ],
+        ]
+        second_loop_body = [
+            '=', [ '[]', 'B', indexB ], [ '[]', 'A', indexA ],
+        ]
+        code_block_parts = [
+            'for', iteration_variable, 1, array_length, first_loop_body,
+            'for', iteration_variable, 1, array_length, second_loop_body,
+        ]
+        code_block = make_block(code_block_parts)
+        lang_table = table(code_block, [ [ 'Basic', 'Pascal', 'Alg' ] ])
+        case = self.rnd.pick([
+            { 'name': 'наибольшим', 'test': 1 },
+            { 'name': 'наименьшим', 'test': -1 },
+        ])
+        self.text = f'''
+Значения двух массивов A[1..{array_length}] и B[1..{array_length}]
+задаются с помощью следующего фрагмента программы: {lang_table}
+Какой элемент массива B будет {case['name']}?'''
+
+        # Analoge of '<=>' operator in Perl.
+        # ToDo: write better implementation.
+        def sign(x: int, y: int) -> int:
+            if x < y:
+                return -1
+            elif x == y:
+                return 0
+            else:
+                return 1
```

Comment:
```suggestion
        def sign(a, b):
            return bool(a > b) - bool(a < b)
```
Можно заменить на что-то такое и вынести в Utils 🤔 
![изображение](https://user-images.githubusercontent.com/24905983/150167682-ea701cb8-a6a8-4426-a4e7-f350b2a7d3bd.png)

Reply:
Согласен, сделал как ты предложил